### PR TITLE
fixes #216. enable linting of root node (used by finalNewline and trailingWhitespace)

### LINKED ISSUE #216: finalNewline and trailingWhitespace linter bugs
Since moving over to PostCSS it doesn't look as though these two linters have been firing.

Their lint functions don't seem to be triggered when using `nodeTypes: ['root']`. Changing this to `nodeTypes: ['rule']`, does fix this but the internal logic for these looks to be incorrect atm.

### PR DESCRIPTION
**Pull request by garethwhittaker**

### CODE REVIEW

File: lib/linter.js [hunk 1]
```diff
@@ -192,6 +170,34 @@ var runChecks = function (ast, fullPath, lines, config) {
...
+        var key = Object.keys(option)[0];
+
+        option[key] = merge(option[key], {
+            line: node.source.start.line
+        });
+
+        lineOptions.push(option);
+    });
+
+    checkNode(ast.root);
```
Comment (jwilsson) [inline lib/linter.js]: Could you add a comment here, explaining why it's necessary (just for future reference).

**Comment by jwilsson** [comment]
One tiny comment. Other than that, LGTM.

**Comment by garethwhittaker** [comment]
updated :smiley:

**Comment by jwilsson** [comment]
Great! Just waiting to see if anyone else has any comments.

**Comment by shellscape** [comment]
I'm good with it.

**Comment by jwilsson** [comment]
Released in `2.0.1` (ping @JoshuaKGoldberg for NuGet package).

**Comment by JoshuaKGoldberg** [comment]
@jwilsson uploaded!

**Pull request merged**